examples/etcd/etcd.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+#
+# This is a wrapper script for the etcd command.
+# This wrapper detects the presence of ETCD_DISCOVERY environment variable and
+# if this variable is set then it will use DNS lookup to collect the IP
+# addresses of the other members of the cluster. This wrapper then adjust the
+# size of the cluster in the discovery service and register itself.
+
+# If we are not running in cluster, then just execute the etcd binary
+if [[ -z "${ETCD_DISCOVERY-}" ]]; then
+  exec /usr/local/bin/etcd "$@"
+fi
+
+# Acquire the IP address of this container
+address=$(getent ahosts ${HOSTNAME} | grep RAW | cut -d ' ' -f 1)
+
+# Wait till this container appears in the DNS subsystem
+endpoints=""
+while true; do
+  sleep 2
+  # FIXME: Remove the uniq (see: origin#2087)
+  endpoints=$(dig etcd A +search +short 2>/dev/null | uniq)
+  if echo ${endpoints} | grep ${address} >/dev/null; then
+    break
+  fi
+done
+
+# Count how many peers we got
+peers_count=0
+for peer in ${endpoints}; do let peers_count++; done
+
+echo "Discovered members: ${endpoints}"
+echo "Adjusting etcd cluster size to ${peers_count}"
+curl -sX PUT ${ETCD_DISCOVERY}/_config/size -d value=${peers_count}
```

Comment:
@mfojtik  It seems that you are updating cluster size here and adding new etcd node. Am I right? If so are you sure it is possible to do, i.e. have you tested it? As far as my understanding goes, you could set cluster size once for boot strapping and after the cluster is booted with that size, you need to add the new etcd node manually.